docs/cohere.md

```diff
@@ -0,0 +1,445 @@
+
+      ```bash
+      psql -d "postgres://<username>:<password>@<host>:<port>/<database-name>" -v cohere_api_key=$COHERE_API_KEY
+      ```
+   Your API key is now available as a psql variable named `cohere_api_key` in your psql session.
+
+   You can also log into the database, then set `cohere_api_key` using the `\getenv` [metacommand](https://www.postgresql.org/docs/current/app-psql.html#APP-PSQL-META-COMMAND-GETENV):
+
+      ```sql
+       \getenv cohere_api_key COHERE_API_KEY
+      ```
+
+4. Pass your API key to your parameterized query:
+    ```sql
+    SELECT cohere_chat_complete
+    ( 'command-r-plus'
+    , 'How much wood would a woodchuck chuck if a woodchuck could chuck wood?'
+    , _api_key=>$1
+    , _seed=>42
+    )->>'text'
+    \bind :cohere_api_key
+    \g
+    ```
+
+   Use [\bind](https://www.postgresql.org/docs/current/app-psql.html#APP-PSQL-META-COMMAND-BIND) to pass the value of `cohere_api_key` to the parameterized query.
+
+   The `\bind` metacommand is available in psql version 16+.
+
+### Handle API keys using pgai from python
+
+1. In your Python environment, include the dotenv and postgres driver packages:
+
+    ```bash
+    pip install python-dotenv
+    pip install psycopg2-binary
+    ```
+
+2. Set your Cohere API key in a .env file or as an environment variable:
+    ```bash
+    COHERE_API_KEY="this-is-my-super-secret-api-key-dont-tell"
+    DB_URL="your connection string"
+    ```
+
+    1. Pass your API key as a parameter to your queries:
+
+        ```python
+        import os
+        from dotenv import load_dotenv
+         
+        load_dotenv()
+        
+        COHERE_API_KEY = os.environ["COHERE_API_KEY"]
+        DB_URL = os.environ["DB_URL"]
+        
+        import psycopg2
+        
+        with psycopg2.connect(DB_URL) as conn:
+            with conn.cursor() as cur:
+                # pass the API key as a parameter to the query. don't use string manipulations
+                cur.execute("""
+                    SELECT cohere_chat_complete
+                    ( 'command-r-plus'
+                    , 'How much wood would a woodchuck chuck if a woodchuck could chuck wood?'
+                    , _api_key=>%s
+                    , _seed=>42
+                    )->>'text'
+                """, (COHERE_API_KEY, ))
+                records = cur.fetchall()
+        ```
+
+       Do not use string manipulation to embed the key as a literal in the SQL query.
+
+## Usage
+
+This section shows you how to use AI directly from your database using SQL.
+
+- [cohere_list_models](#cohere_list_models)
+- [cohere_tokenize](#cohere_tokenize)
+- [cohere_detokenize](#cohere_detokenize)
+- [cohere_embed](#cohere_embed)
+- [cohere_classify](#cohere_classify)
+- [cohere_classify_simple](#cohere_classify_simple)
+- [cohere_rerank](#cohere_rerank)
+- [cohere_rerank_simple](#cohere_rerank_simple)
+- [cohere_chat_complete](#cohere_chat_complete)
+
+### cohere_list_models
+
```

Comment:
There are multiple options here, can you update to match the openAI page and be a list of options please. Something along the lines of:

You can list the AI models supported by Cohere in the the following ways:

* List the models supported by the Cohere platform:
   ```
    indented code example
    ```
    Result:
    ```
    indented result
    ```

* Next item

Reply:
done